common/lib/xmodule/xmodule/tests/test_capa_module.py

```diff
@@ -700,6 +700,42 @@ def test_submit_problem_error(self):
             # Expect that the number of attempts is NOT incremented
             self.assertEqual(module.attempts, 1)
 
+    def test_submit_problem_error_with_codejail_exception(self):
+
+        # Try each exception that capa_module should handle
+        exception_classes = [StudentInputError,
+                             LoncapaProblemError,
+                             ResponseError]
+        for exception_class in exception_classes:
+
+            # Create the module
+            module = CapaFactory.create(attempts=1)
+
+            # Ensure that the user is NOT staff
+            module.system.user_is_staff = False
+
+            # Simulate a codejail exception 'Exception: test error'
+            with patch('capa.capa_problem.LoncapaProblem.grade_answers') as mock_grade:
+                try:
+                    raise ResponseError(
+                        'Couldn\'t execute jailed code: stdout: \'\', '
+                        'stderr: \'Traceback (most recent call last):\\n'
+                        '  File "jailed_code", line 15, in <module>\\n'
+                        '    exec code in g_dict\\n  File "<string>", line 67, in <module>\\n'
+                        '  File "<string>", line 65, in check_func\\n'
+                        'Exception: test error\\n\' with status code: 1',)
```

Comment:
@pdpinch so this code us for exceptions that have new line `\\n` in its message
**code** `msg = inst.args[0].split("\\n")[-2].split(": ", 1)[1] ` 

When there is not new line in exception text `Exception: test error with status code: 1` it gives error `Error: test error`